verifiers/internal/gha/bundle.go

```diff
@@ -156,6 +164,41 @@ func matchRekorEntryWithEnvelope(tlogEntry *v1.TransparencyLogEntry, env *dsseli
 	return nil
 }
 
+// matchRekorEntryWithEnvelopeDSSEv001 handles matchRekorEntryWithEnvelope for the dsse v0.0.1 type version.
+func matchRekorEntryWithEnvelopeDSSEv001(tlogEntry *v1.TransparencyLogEntry, env *dsselib.Envelope) error {
+	canonicalBody := tlogEntry.GetCanonicalizedBody()
+	var dsseObj models.DSSE
+	if err := json.Unmarshal(canonicalBody, &dsseObj); err != nil {
+		return fmt.Errorf("%w: %s", ErrorUnexpectedEntryType, err)
+	}
+	var dsseSchemaObj models.DSSEV001Schema
+	specMarshal, err := json.Marshal(dsseObj.Spec)
```

Comment:
this feels odd? are we using json as an intermediate layer while converting between objects? I may not know enough about to go to understand if this is normal though.

Reply:
Yeah, the docs say that the verifier has to parse out the objects to match the sigatures

 - https://pkg.go.dev/github.com/sigstore/protobuf-specs/gen/pb-go/rekor/v1#TransparencyLogEntry
 
And then If you see the test cases for `CanonicalizedBody`, the objects with the signaures are wrapped within another object.

 - github.com/sigstore/rekor/pkg/generated/model

But then the "Spec"'s `DSSESchema` is just an interface with no methods, so we have to marshall that again to get the data.

 - https://pkg.go.dev/github.com/sigstore/rekor@v1.3.6/pkg/generated/models#DSSESchema
 - https://pkg.go.dev/github.com/sigstore/rekor@v1.3.6/pkg/generated/models#DSSEV001Schema